skus/service.go

```diff
@@ -1304,6 +1304,56 @@ func (s *Service) RunStoreSignedOrderCredentialsJob(ctx context.Context) (bool,
 	}
 }
 
+// verifyDeveloperNotification - verify the developer notification from playstore
+func (s *Service) verifyDeveloperNotification(ctx context.Context, dn *DeveloperNotification) error {
+	if dn == nil || dn.SubscriptionNotification.PurchaseToken == "" {
+		return errors.New("notification has no purchase token")
+	}
+
+	// lookup the order based on the token as externalID
+	o, err := s.Datastore.GetOrderByExternalID(dn.SubscriptionNotification.PurchaseToken)
+	if err != nil {
+		return fmt.Errorf("failed to get order from db: %w", err)
+	}
+	// have order, now validate the receipt from the notification
+	_, err = s.validateReceipt(ctx, &o.ID, SubmitReceiptRequestV1{
```

Comment:
could o.ID be potential nil pointer if get order by external id returns sql no rows

Reply:
https://github.com/brave-intl/bat-go/pull/1558/commits/1f332ef8be68974360f74e91c43396a05bb03b52